packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -100,13 +100,26 @@ export interface ISummarizerEvents extends IEvent {
      */
     (event: "summarizingError", listener: (error: ISummarizingWarning) => void);
 }
+export type SummarizerStopReason =
+    /** Summarizer client failed to summarize in all 3 consecutive attempts. */
+    | "failToSummarize"
+    /** Highly unexpected error encountered while responding to a summary ack error. */
+    | "handleSummaryAckFatalError"
+    /** Summarizer client detected that its parent is no longer elected the summarizer. */
+    | "parentNoLongerSummarizer"
+    /** Parent client reported that it is no longer connected. */
+    | "parentNotConnected"
+    /** Parent client reported that it is no longer elected the summarizer. */
+    | "parentShouldNotSummarize"
```

Comment:
Strictly looking at this list, it's not clear what's the difference between these two cases:

/** Summarizer client detected that its parent is no longer elected the summarizer. */ 
| "parentNoLongerSummarizer"

/** Parent client reported that it is no longer elected the summarizer. */
| "parentShouldNotSummarize"

Isn't it one and the same? If not, some clarification would be great!

Reply:
The comments explain the difference, but it's not clear from the message themselves:
- "parentShouldNotSummarize" is reported by the summarizer client (child client)
- "parentNoLongerSummarizer" is reported by the parent client

We could probably give these better names, or even unify them, but I didn't bother since the one reported by the summarizer client probably needs to be revisited anyway.

Reply:
Looking at it in telemetry - what difference can I assume between the two? I.e. in both cases I'd guess it's because parent client disconnected at some point and is no longer oldest client, right?
I guess I'm not sure what kind of message are we sending trying to differentiate these cases. Maybe there is one, but it's not clear from these names / descriptions.

Reply:
Well it's just a "more information is better" type of thing. Which client realized it first? I'd expect it to be the parent client, but perhaps if something is wrong somewhere, we could help disambiguate it via this distinction.

Reply:
That's fine, but it would be great to have some description on expectations (to help future OCEs realize what is normal and what is not and likely needs to look deeper).

For example, my naïve interpretations is something like that:
parentShouldNotSummarize should not happen that often because in majority of the cases we would first go through parentNotConnected and that would stop summarizer. But if reconnection happens quickly, we may find ourselves skipping parentNotConnected and stopping summarizer with parentShouldNotSummarize reason.
The difference between parentShouldNotSummarize and parentNoLongerSummarizer is roughly this ...

Basically I want to be able to map what is observed in telemetry to how code is expected to work and be able to spot anomalies (or earliest signals that something started to go wrong). It's rather hard to do as code is rather complicated, so any help in the form of inline documentation helps.